casper/src/main/resources/Registry.rho

```diff
@@ -64,8 +64,10 @@ in {
                   true => {
                     ops!("buildUri", *pubKeyBytes, *uriCh) |
                     for (@uri <- uriCh; @map <- _registryStore) {
-                      _registryStore!(map.set(uri, (nonce, data))) |
-                      ret!(uri)
+                      match map.get(uri) {
+                        Nil => { _registryStore!(map.set(uri, (nonce, data))) | ret!(uri) }
+                        (oldNonce, _) => { if (nonce > oldNonce) { _registryStore!(map.set(uri, (nonce, data))) | ret!(uri) } else ret!(Nil) }
+                      }
```

Comment:
Please squash this into the original `registry.rho` commit

Reply:
Done